test/unit/Chainweb/Test/Pact5/PactServiceTest.hs

```diff
@@ -335,15 +321,17 @@ testMempoolExcludesInvalid baseRdb = runResourceT $ do
                 ]
             }
 
+        badChain <- buildCwCmd v $ transferCmd 1.0 & set cbChainId (unsafeChainId 1)
+
         let pact4Hash = Pact5.Hash . Pact4.unHash . Pact4.toUntypedHash . Pact4._cmdHash
-        _ <- advanceAllChains fixture $ onChain cid $ \ph pactQueue mempool -> do
-            insertMempool mempool CheckedInsert [regularTx1]
+        _ <- advanceAllChains fixture $ onChain chain0 $ \ph pactQueue mempool -> do
+            mempoolInsert5 mempool CheckedInsert [regularTx1]
             bip <- throwIfNotPact5 =<< throwIfNoHistory =<< newBlock noMiner NewBlockFill (ParentHeader ph) pactQueue
             return $ finalizeBlock bip
 
-        _ <- advanceAllChains fixture $ onChain cid $ \ph pactQueue mempool -> do
-            mempoolInsert mempool CheckedInsert $ Vector.fromList [badParse, badSigs]
-            insertMempool mempool CheckedInsert [badUnique, badFuture, badPast, badTxHash]
+        _ <- advanceAllChains fixture $ onChain chain0 $ \ph pactQueue mempool -> do
+            mempoolInsert mempool UncheckedInsert $ Vector.fromList [badParse, badSigs]
+            mempoolInsert5 mempool UncheckedInsert [badChain, badUnique, badFuture, badPast, badTxHash]
```

Comment:
Double checking that this change in semantic doesn't change the intent of the test. Do we want unchecked or checked here?

Reply:
A checked insert will test the validation logic of the mempool, but here we want to test that NewBlock behaves correctly even if the mempool is wrong.